dds/idl/marshal_generator.cpp

```diff
@@ -2476,16 +2476,16 @@ namespace {
             "      case " << id << ": {\n"
             "        if (!" << generate_field_stream(
             indent, field, ">> stru" + value_access, wrap_nested_key_only, intro) << ") {\n";
-          AST_Type* const field_type = resolveActualType(field->field_type());
-          const Classification fld_cls = classify(field_type);
+          AST_Type* const field_type2 = resolveActualType(field->field_type());
```

Comment:
I would suggest renaming the function argument `field_filter` and leave this as `field_type`.